prog.go

```diff
@@ -155,14 +155,20 @@ func (p *BPFProg) SetAttachTarget(attachProgFD int, attachFuncName string) error
 	return nil
 }
 
-// TODO: fix API to return error
-func (p *BPFProg) SetProgramType(progType BPFProgType) {
-	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
+func (p *BPFProg) SetProgramType(progType BPFProgType) error {
```

Comment:
To avoid unforeseen breaking changes, set the old API as `// Deprecated` (like [in](https://github.com/aquasecurity/libbpfgo/blob/d3ea3baf9164fa237f7f1b3e7df3850681370b90/link.go#L84)) and create the new one aligning the names, e.g.:

`func (p *BPFProg) SetType(progType BPFProgType) error {`
`func (p *BPFProg) SetExpectedAttachType(attachType BPFAttachType) error {`